File: maths-pattern/even_odd.c
#include <stdbool.h>
#include <stdio.h>

bool is_even(int n) {
  if ((n & 1) == 0)
    return true;
  else
    return false;
}

int main(int argc, char *argv[]) {

  int n = 9998;
  printf("is even or odd?: %d\n", is_even(n));

  return 0;
}

File: maths-pattern/even_odd.rs
fn is_even(n: i8) -> bool {
    if (n & 1) == 0 {
        return true;
    } else {
        return false;
    }
}

fn main() {
    let num: i8 = 98;
    println!("is even?: {}", is_even(num));
}
